src/sqlfluff/core/templaters/dbt.py

```diff
@@ -151,24 +154,41 @@ def _get_profiles_dir(self):
         """
         from dbt.config.profile import PROFILES_DIR
 
-        return os.path.expanduser(
-            self.sqlfluff_config.get_section(
-                (self.templater_selector, self.name, "profiles_dir")
+        dbt_profiles_dir = os.path.abspath(
+            os.path.expanduser(
+                self.sqlfluff_config.get_section(
+                    (self.templater_selector, self.name, "profiles_dir")
+                )
+                or PROFILES_DIR
             )
-            or PROFILES_DIR
         )
 
+        if not os.path.exists(dbt_profiles_dir):
+            templater_logger.error(
+                f"dbt_profiles_dir: {dbt_profiles_dir} could not be accessed. Check it exists."
+            )
+
+        return dbt_profiles_dir
+
     def _get_project_dir(self):
         """Get the dbt project directory from the configuration.
 
         Defaults to the working directory.
         """
-        return os.path.expanduser(
-            self.sqlfluff_config.get_section(
-                (self.templater_selector, self.name, "project_dir")
+        dbt_project_dir = os.path.abspath(
+            os.path.expanduser(
+                self.sqlfluff_config.get_section(
+                    (self.templater_selector, self.name, "project_dir")
+                )
+                or os.getcwd()
             )
-            or os.getcwd()
         )
+        if not os.path.exists(dbt_project_dir):
+            templater_logger.error(
```

Comment:
codecov complaining too ^^

Actually I wonder, should we be actually throwing an Exception here ? I don't think there's much we can do if the dbt project directory isn't found right?

Reply:
@dmateusp I initially tried throwing an `IOError` exception here but this was handled in [commands.py](https://github.com/sqlfluff/sqlfluff/blob/master/src/sqlfluff/cli/commands.py#L353) with a custom (not relevant) error message, so the message I wanted to show here never gets shown.

As the code is now we will get an exception later anyway when we try to change directory into a non existing directory, but I agree an exception here would be more appropriate. Any suggestions for how to get round `commands.py `obscuring the `IOError `exception? Should we change that code or add a new exception class here e.g. `ConfigParsingError`?

I've added a test to cover these lines [here](https://github.com/sqlfluff/sqlfluff/pull/898/files#diff-5d3be0ba02e85767f3906e01bdb3611ce1c9defc129ba33dde3c2ecd6e6727e0R225)

Reply:
I think you could create an exception like `DbtProjectFolderDoesNotExistError` (adding a new exception class like you suggested), then the `except OSError` won't handle it anymore and you can add another except clause to catch this specific exception

but now that we have coverage and logging I'm happy either way